alacritty/src/event.rs

```diff
@@ -381,16 +384,37 @@ impl<N: Notify + OnResize> Processor<N> {
             config,
             message_buffer,
             display,
+            event_queue: Vec::new(),
+        }
+    }
+
+    #[inline]
+    /// Return `true` if `event_queue` is empty, `true` otherwise.
+    fn event_queue_empty(&mut self) -> bool {
+        #[cfg(not(any(target_os = "macos", windows)))]
+        {
+            let wayland_event_queue = match self.display.wayland_event_queue.as_mut() {
+                Some(wayland_event_queue) => wayland_event_queue,
+                None => return self.event_queue.is_empty(),
+            };
+
+            // On Wayland we also dispatch and check its own event queue
+            let events_dispatched = wayland_event_queue
+                .dispatch_pending(&mut (), |_, _, _| {})
+                .expect("failed to dispatch event queue");
+
+            self.event_queue.is_empty() && events_dispatched == 0
         }
+
+        #[cfg(any(target_os = "macos", windows))]
+        self.event_queue.is_empty()
```

Comment:
I think we should just duplicate the entire method. I usually find it much easier to read than having different blocks inside of it.

That way we don't need the extra empty block just for the config attribute.